symfony/src/Controller/DeceasedReportsController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+namespace App\Controller;
+
+use App\Entity\DeceasedReport;
+use App\Form\DeceasedReportReviewType;
+use App\Form\DeceasedReportType;
+use App\Service\DeceasedReportsService;
+use App\Service\ParticipantSummaryService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/s/deceased-reports")
+ */
+class DeceasedReportsController extends AbstractController
+{
+    /**
+     * @Route("/", name="deceased_reports_index")
+     */
+    public function participantObservationIndex(Request $request, SessionInterface $session, DeceasedReportsService $deceasedReportsService) {
+        $statusFilter = $request->query->get('status', 'preliminary');
+        $organizationId = ($request->query->get('all', false) === false) ? $session->get('siteOrganizationId', null) : null;
+        $reports = $deceasedReportsService->getDeceasedReports($organizationId, $statusFilter);
+        $reports = $this->formatReportTableRows($reports);
+        return $this->render('deceasedreports/index.html.twig', [
+            'statusFilter' => $statusFilter,
+            'reports' => $reports
+        ]);
+    }
+
+    /**
+     * @Route("/{participantId}/{reportId}", name="deceased_report_review", requirements={"participantId"="P\d+","reportId"="\d+"})
+     */
+    public function deceasedReportReview(Request $request, ParticipantSummaryService $participantSummaryService, DeceasedReportsService $deceasedReportsService, $participantId, $reportId) {
+        $participant = $participantSummaryService->getParticipantById($participantId);
+        if (!$participant) {
+            throw $this->createNotFoundException('Participant not found.');
+        }
+        $fhirData = $deceasedReportsService->getDeceasedReportById($participantId, $reportId);
+        $report = (new DeceasedReport())->loadFromFhirObservation($fhirData);
+        if ($report->getSubmittedBy() == $this->getUser()->getEmail() && $report->getReportStatus() == 'preliminary') {
```

Comment:
Can we change these (and several other `==` comparisons throughout the PHP code) to strict comparisons?

Reply:
Fixed in 11f84c8